packages/contracts/test/forwarder/forwarder.ts

```diff
@@ -0,0 +1,306 @@
+            takerFee: ZeroEx.toBaseUnitAmount(new BigNumber(0), DECIMALS_DEFAULT),
+        };
+        const privateKey = constants.TESTRPC_PRIVATE_KEYS[accounts.indexOf(makerAddress)];
+        orderFactory = new OrderFactory(privateKey, defaultOrderParams);
+
+        const forwarderArgs = [
+            exchangeInstance.address,
+            etherTokenInstance.address,
+            zrxToken.address,
+            AssetProxyId.ERC20,
+        ];
+        const forwarderInstance = await deployer.deployAndSaveAsync(ContractName.Forwarder, forwarderArgs);
+        forwarderContract = new ForwarderContract(forwarderInstance.abi, forwarderInstance.address, provider);
+        await forwarderContract.setERC20ProxyApproval.sendTransactionAsync(AssetProxyId.ERC20, { from: owner });
+        forwarderWrapper = new ForwarderWrapper(forwarderContract, web3Wrapper);
+        erc20Wrapper.addTokenOwnerAddress(forwarderInstance.address);
+
+        web3Wrapper.abiDecoder.addABI(forwarderContract.abi);
+        web3Wrapper.abiDecoder.addABI(exchangeInstance.abi);
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+        erc20Balances = await erc20Wrapper.getBalancesAsync();
+        signedOrder = orderFactory.newSignedOrder();
+        feeOrder = orderFactory.newSignedOrder({
+            makerAssetData: assetProxyUtils.encodeERC20ProxyData(zrxToken.address),
+            takerFee: ZeroEx.toBaseUnitAmount(new BigNumber(1), DECIMALS_DEFAULT),
+        });
+        orderWithFees = orderFactory.newSignedOrder({
+            takerFee: ZeroEx.toBaseUnitAmount(new BigNumber(1), DECIMALS_DEFAULT),
+        });
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    describe('buyTokens', () => {
+        it('should fill the order', async () => {
+            const fillAmount = signedOrder.takerAssetAmount.div(2);
+            const makerBalanceBefore = erc20Balances[makerAddress][defaultMakerAssetAddress];
+            const takerBalanceBefore = erc20Balances[takerAddress][defaultMakerAssetAddress];
+            const tx = await forwarderWrapper.buyTokensAsync([signedOrder], [], fillAmount, takerAddress);
+            const newBalances = await erc20Wrapper.getBalancesAsync();
+            const makerBalanceAfter = newBalances[makerAddress][defaultMakerAssetAddress];
+            const takerBalanceAfter = newBalances[takerAddress][defaultMakerAssetAddress];
+            const makerTokenFillAmount = fillAmount
+                .times(signedOrder.makerAssetAmount)
+                .dividedToIntegerBy(signedOrder.takerAssetAmount);
+
+            expect(makerBalanceAfter).to.be.bignumber.equal(makerBalanceBefore.minus(makerTokenFillAmount));
+            expect(takerBalanceAfter).to.be.bignumber.equal(takerBalanceBefore.plus(makerTokenFillAmount));
+            expect(newBalances[forwarderContract.address][weth.address]).to.be.bignumber.equal(new BigNumber(0));
+        });
+
```

Comment:
Remove new line. Why is TSLint not catching this?

Reply:
I believe our tslint only errors on multiple consecutive lines, rather than breaking up with single lines to aid readability.

Reply:
I see... We should change that.